src/providers/aws.rs

```diff
@@ -169,19 +169,15 @@ impl MachineSetup<YesAmi> {
 /// Each individual region is handled by `AWSRegion`.
 pub struct AWSLauncher<P = DefaultCredentialsProvider> {
     credential_provider: Box<dyn Fn() -> Result<P, Error>>,
-    max_instance_duration: std::time::Duration,
+    max_instance_duration_hours: i64,
```

Comment:
I'd probably just use `isize` here since we don't care too much about its range.

Reply:
Also, should it be `usize` since the value can't be negative?